server/remote_cache/byte_stream_client/byte_stream_client.go

```diff
@@ -169,26 +172,59 @@ func stripUser(u *url.URL) *url.URL {
 	return &copy
 }
 
-func grpcTargetForFileURL(u *url.URL, grpcs bool) string {
+func getTargetForURL(u *url.URL, grpcs bool) string {
 	target := url.URL{Scheme: "grpc", User: u.User, Host: u.Host}
 	if grpcs {
 		target.Scheme = "grpcs"
+		if u.Port() == "" {
+			target.Host = u.Hostname() + ":443"
+		}
+	} else if u.Port() == "" {
+		target.Host = u.Hostname() + ":80"
 	}
+
 	return target.String()
 }
 
-func streamFromUrl(ctx context.Context, env environment.Env, url *url.URL, grpcs bool, offset int64, limit int64, writer io.Writer) error {
-	if url.Port() == "" && grpcs {
-		url.Host = url.Hostname() + ":443"
-	} else if url.Port() == "" {
-		url.Host = url.Hostname() + ":80"
+func getCachedGrpcClientConnPool(env environment.Env, target string) (grpc.ClientConnInterface, error) {
+	if cc := env.GetGrpcClientConnPoolCache(); cc != nil {
+		return cc.GetGrpcClientConnPoolForURL(target)
 	}
+	return nil, nil
+}
 
-	conn, err := grpc_client.DialInternal(env, grpcTargetForFileURL(url, grpcs))
+func isPermittedForDial(target string) bool {
+	u, err := url.Parse(target)
 	if err != nil {
-		return err
+		return false
+	}
+	if cache_api_url.String() == "" {
+		return true
+	}
+
+	return u.Hostname() == "localhost" || (urlutil.GetDomain(u.Hostname()) == urlutil.GetDomain(cache_api_url.WithPath("").Hostname()))
+}
+
+func streamFromUrl(ctx context.Context, env environment.Env, url *url.URL, grpcs bool, offset int64, limit int64, writer io.Writer) (err error) {
+	target := getTargetForURL(url, grpcs)
+	var conn grpc.ClientConnInterface
+	if grpc_client.PoolCacheEnabled() {
```

Comment:
use a different flag here instead of depending on pool caching being enabled?

Reply:
Done